### ISSUE TITLE
docker: nginx gzip settings

### ISSUE THREAD

**Issue by slint** [zenodo/zenodo#667]
The current nginx configuration is limited and doesn't include all MIME types that Zenodo serves. 

Another issue comes with resources that include ETags in their response headers for browser caching. Because nginx (from v1.7.x) considers gzipped responses technically as "modified" content, it downgrades strong ETags to weak (aka. prepends them with "W/"), which causes the etag check to fail on future requests.

- [ ] Add settings such as compression level and gzip-related headers.
- [ ] Find and include in `gzip_types` all MIME types that Zenodo uses in responses that should be compressed.
- [ ] Test that resources with ETags, properly return `304 Not Modified` when future requests with proper `If-Match` HTTP headers and weak ETags are used.

**Comment by lnielsen** [comment]
Is the state of this ticket correct?

**Comment by slint** [comment]
Yes, this wasn't tested thoroughly. Most of the cases/mimetypes are covered though
